tests/unittest/_torch/speculative/hw_agnostic/test_mtp.py

```
@@ -1733,6 +1741,69 @@ def test_prepare_drafter_inputs(
+@pytest.mark.parametrize(
+    ("architecture", "one_model", "expected"),
+    [
+        ("Gemma4ForCausalLM", True, True),
+        ("Gemma4ForConditionalGeneration", False, False),
+        ("LlamaForCausalLM", True, False),
+    ],
+)
+def test_mtp_shared_kv_config(architecture, one_model, expected):
+    spec_config = MTPDecodingConfig(
+        max_draft_len=3,
+        speculative_model="/tmp/assistant",
```

Ruff (0.16.0)

[error] Probable insecure usage of temporary file or directory: "/tmp/assistant"

(S108)